# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/SourceSurfaceSkia.cpp
# Commit: e5570884e8e4
# Full Hash: e5570884e8e43505e1a35ce3fa7ab8fc33ede79d
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-04-16 09:48:54
# Regressor Bug: 1821512
# File Overlap Count: 1
# Description:
#   Bug 1821512 - Fix Skia usage for API changes. r=jrmuizel
#   
#   This just tries to address fairly random changes in the Skia API and correct
#   our usage of it in Moz2D and some other places.
#   
# ==============================================================================

diff -r efc92c86fafd -r e5570884e8e4 gfx/2d/SourceSurfaceSkia.cpp
--- a/gfx/2d/SourceSurfaceSkia.cpp	Sun Apr 16 03:59:59 2023 +0000
+++ b/gfx/2d/SourceSurfaceSkia.cpp	Sun Apr 16 04:00:00 2023 +0000
@@ -11,6 +11,7 @@
 #include "skia/include/core/SkData.h"
 #include "skia/include/core/SkImage.h"
 #include "skia/include/core/SkSurface.h"
+#include "skia/include/private/base/SkMalloc.h"
 #include "mozilla/CheckedInt.h"
 
 namespace mozilla::gfx {
@@ -134,7 +135,10 @@
   } else if (aFormat != SurfaceFormat::UNKNOWN) {
     mFormat = aFormat;
     SkImageInfo info = MakeSkiaImageInfo(mSize, mFormat);
-    mStride = SkAlign4(info.minRowBytes());
+    mStride = GetAlignedStride<4>(info.width(), info.bytesPerPixel());
+    if (!mStride) {
+      return false;
+    }
   } else {
     return false;
   }
@@ -154,7 +158,10 @@
     return nullptr;
   }
   SkImageInfo info = MakeSkiaImageInfo(aRect.Size(), mFormat);
-  size_t stride = SkAlign4(info.minRowBytes());
+  size_t stride = GetAlignedStride<4>(info.width(), info.bytesPerPixel());
+  if (!stride) {
+    return nullptr;
+  }
   sk_sp<SkImage> subImage = ReadSkImage(mImage, info, stride, aRect.x, aRect.y);
   if (!subImage) {
     return nullptr;
